src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php

```diff
@@ -138,13 +147,40 @@ public function deleteAction(int $stateId): RedirectResponse
      */
     public function editAction(int $stateId, Request $request): Response
     {
-        return $this->redirect(
-            $this->getAdminLink(
-                'AdminStates',
-                ['updatestate' => '', 'id_state' => $stateId],
-                true
-            )
-        );
+        try {
+            /** @var EditableState $editableState */
+            $editableState = $this->getQueryBus()->handle(new GetStateForEditing((int) $stateId));
+        } catch (StateException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_states_index');
+        }
+
+        $stateForm = null;
+
+        try {
+            $stateForm = $this->get('prestashop.core.form.identifiable_object.builder.state_form_builder')->getFormFor((int) $stateId);
```

Comment:
```suggestion
            $stateForm = $this->getStateFormBuilder()->getFormFor((int) $stateId);
```

Reply:
You could create private methods to get the builder and handler services. But it's not mandatory, just a suggestion.

Reply:
@jolelievre & @atomiix Fixed :)